### PR TITLE
feat(badge): functional component

### PR DESCRIPTION
**Pull request by alexsasharegan**
- badge as functional component
- updates badge tests
- updates test helpers
- adds vue-functional-data-merge
  - util for accurately merging component defined data with parent applied data

### CODE REVIEW

**Comment by alexsasharegan** [comment]
I'll need the changes here before I can start applying other components.

File: package.json [hunk 1]
```diff
@@ -63,10 +63,11 @@
   },
   "dependencies": {
     "bootstrap": "^4.0.0-beta",
+    "jquery": "jquery@>=3.0.0",
```
Comment (tmorehouse) [inline package.json]: I didn't realize we have a jQuery dependency now...

Comment (alexsasharegan) [inline package.json]: Yeah... when did that get in there? Is that in master?

Comment (tmorehouse) [inline package.json]: Might be in there for a temp solution for something.

Comment (tmorehouse) [inline package.json]: not in `dev` branch either.  only 1.x branch

Comment (alexsasharegan) [inline package.json]: Say bye-bye 👋

**Comment by tmorehouse** [comment]
NO complaints with jQuery gone....

**Comment by alexsasharegan** [comment]
image

**Comment by tmorehouse** [comment]
So yeah, maybe put it back in for now...

**Comment by alexsasharegan** [comment]
How are you guys merging these? Squash and merge?

**Comment by tmorehouse** [comment]
Yep. Squash & merge

**Pull request merged**